src/plugins/discover/public/application/embeddable/search_embeddable.ts

```diff
@@ -389,6 +389,11 @@ export class SearchEmbeddable
     if (forceFetch || isFetchRequired) {
       this.filtersSearchSource!.setField('filter', this.input.filters);
       this.filtersSearchSource!.setField('query', this.input.query);
+      if (this.input.query?.query || this.input.filters?.length) {
+        this.filtersSearchSource!.setField('highlightAll', true);
+      } else {
+        this.filtersSearchSource!.removeField('highlightAll');
+      }
```

Comment:
It looks like you've made the highlighting conditional, but the Discover app does it unconditionally. Why the difference?

Reply:
I don't think we need it once the filter/query is removed, feels cleaner this way